src/core/public/chrome/ui/header/recent_items.tsx

```diff
@@ -2,76 +2,164 @@
  * Copyright OpenSearch Contributors
  * SPDX-License-Identifier: Apache-2.0
  */
-import React, { useMemo, useState } from 'react';
+import React, { useMemo, useState, useEffect } from 'react';
 import * as Rx from 'rxjs';
+import moment from 'moment';
 import {
-  EuiPopover,
-  EuiTextColor,
+  EuiPanel,
   EuiListGroup,
   EuiListGroupItem,
   EuiTitle,
+  EuiPopoverTitle,
+  EuiIcon,
   EuiText,
+  EuiButtonEmpty,
   EuiSpacer,
   EuiHeaderSectionItemButtonProps,
   EuiButtonIcon,
+  EuiRadioGroup,
+  EuiTextColor,
+  EuiPopover,
   EuiToolTip,
 } from '@elastic/eui';
 import { i18n } from '@osd/i18n';
 import useObservable from 'react-use/lib/useObservable';
-import { ChromeRecentlyAccessedHistoryItem } from '../..';
+import { SavedObjectsNamespaceType } from 'src/core/public';
+import { ChromeRecentlyAccessedHistoryItem, SavedObject } from 'opensearch-dashboards/public';
 import { WorkspaceObject } from '../../../workspace';
-import { createRecentNavLink } from './nav_link';
 import { HttpStart } from '../../../http';
-import { ChromeNavLink } from '../../../';
 import './recent_items.scss';
 
+const widthForRightMargin = 8;
+
 export interface Props {
   recentlyAccessed$: Rx.Observable<ChromeRecentlyAccessedHistoryItem[]>;
   workspaceList$: Rx.Observable<WorkspaceObject[]>;
   navigateToUrl: (url: string) => Promise<void>;
-  basePath: HttpStart['basePath'];
-  navLinks$: Rx.Observable<ChromeNavLink[]>;
   renderBreadcrumbs: React.JSX.Element;
   buttonSize?: EuiHeaderSectionItemButtonProps['size'];
+  http: HttpStart;
+}
+
+interface SavedObjectMetadata {
+  icon?: string;
+  title?: string;
+  editUrl?: string;
+  inAppUrl?: { path: string; uiCapabilitiesPath: string };
+  namespaceType?: SavedObjectsNamespaceType;
 }
+export type SavedObjectWithMetadata<T = unknown> = SavedObject<T> & {
+  meta: SavedObjectMetadata;
+};
+
+type DetailedRecentlyAccessedItem = ChromeRecentlyAccessedHistoryItem &
+  SavedObjectWithMetadata &
+  ChromeRecentlyAccessedHistoryItem['meta'] & {
+    updatedAt: number;
+    workspaceName?: string;
+  };
+
+const bulkGetDetail = (savedObjects: Array<Pick<SavedObject, 'type' | 'id'>>, http: HttpStart) => {
+  return Promise.all(
+    savedObjects.map((obj) =>
+      http
+        .get<SavedObjectWithMetadata>(
+          `/api/opensearch-dashboards/management/saved_objects/${encodeURIComponent(
+            obj.type
+          )}/${encodeURIComponent(obj.id)}`
+        )
+        .catch((error) => ({
+          id: obj.id,
+          type: obj.type,
+          error,
+          attributes: {},
+          references: [],
+          meta: {},
+          updated_at: '',
+        }))
+    )
+  );
+};
+
+const recentsRadios = [
+  {
+    id: '5',
+    label: '5 pages',
+  },
+  {
+    id: '10',
+    label: '10 pages',
+  },
+  {
+    id: '15',
+    label: '15 pages',
+  },
+];
```

Comment:
```suggestion
const recentsRadios = [
  {
    id: 5,
    label: '5 pages',
  },
  {
    id: 10,
    label: '10 pages',
  },
  {
    id: 15,
    label: '15 pages',
  },
];
```

Nit: number seems to make more sense to me.

Reply:
Thanks, get it.